browser/extensions/api/brave_rewards_api.cc

```diff
@@ -164,6 +201,51 @@ void BraveRewardsTipTwitterUserFunction::OnTwitterPublisherInfoSaved(
   Release();
 }
 
+void BraveRewardsTipRedditUserFunction::OnRedditPublisherInfoSaved(
+    std::unique_ptr<::brave_rewards::ContentSite> publisher_info) {
+  std::unique_ptr<brave_rewards::TipRedditUser::Params> params(
+      brave_rewards::TipRedditUser::Params::Create(*args_));
+
+  if (!publisher_info) {
+    Release();
+    return;
+  }
+
+  content::WebContents* contents = nullptr;
+  if (!ExtensionTabUtil::GetTabById(
+          params->tab_id,
+          Profile::FromBrowserContext(browser_context()),
+          false,
+          nullptr,
+          nullptr,
+          &contents,
+          nullptr)) {
+      return;
+  }
+
+  std::unique_ptr<base::DictionaryValue> params_dict =
```

Comment:
don't create with ptr

Reply:
Until `OpenTipDialog` is refactored, this is what has to be passed in.